ivadomed/adaptative.py

```diff
@@ -0,0 +1,695 @@
+    It works on RAM or on the fly and can be saved for later.
+    """
+
+    def __init__(self, hdf5, contrasts, path, target_suffix=None, roi_suffix=None,
+                 slices=False, dim=2):
+        """
+        Initialize the Dataframe
+        """
+        # Number of dimension
+        self.dim = dim
+        # List of all contrasts
+        self.contrasts = copy.deepcopy(contrasts)
+
+        if target_suffix:
+            for gt in target_suffix:
+                self.contrasts.append('gt/' + gt)
+        else:
+            self.contrasts.append('gt')
+
+        if roi_suffix:
+            for roi in roi_suffix:
+                self.contrasts.append('roi/' + roi)
+        else:
+            self.contrasts.append('ROI')
+
+        self.df = None
+        self.filter = slices
+
+        # Data frame
+        if os.path.exists(path):
+            self.load_dataframe(path)
+        else:
+            self.create_df(hdf5)
+
+    def shuffle(self):
+        """Shuffle the whole data frame"""
+        self.df = self.df.sample(frac=1)
+
+    def load_dataframe(self, path):
+        """
+        Load the dataframe from a csv file.
+        """
+        try:
+            self.df = pd.read_csv(path)
+            print("Dataframe has been correctly loaded from {}.".format(path))
+        except FileNotFoundError:
+            print("No csv file found")
+
+    def save(self, path):
+        """
+        Save the dataframe into a csv file.
+        """
+        try:
+            self.df.to_csv(path, index=False)
+            print("Dataframe has been saved at {}.".format(path))
+        except FileNotFoundError:
+            print("Wrong path.")
+
+    def create_df(self, hdf5):
+        """
+        Generate the Data frame using the hdf5 file
+        """
+        # Template of a line
+        empty_line = {col: 'None' for col in self.contrasts}
+        empty_line['Slices'] = 'None'
+
+        # Initialize the data frame
+        col_names = [col for col in empty_line.keys()]
+        col_names.append('Subjects')
+        df = pd.DataFrame(columns=col_names)
+        print(hdf5.attrs['patients_id'])
+        # Filling the data frame
+        for subject in hdf5.attrs['patients_id']:
+            # Getting the Group the corresponding patient
+            print(subject)
+            grp = hdf5[subject]
+            line = copy.deepcopy(empty_line)
+            line['Subjects'] = subject
+
+            # inputs
+            assert 'inputs' in grp.keys()
+            inputs = grp['inputs']
+            for c in inputs.attrs['contrast']:
+                if c in col_names:
+                    line[c] = '{}/inputs/{}'.format(subject, c)
+                else:
+                    continue
+            # GT
+            assert 'gt' in grp.keys()
+            inputs = grp['gt']
+            print("input = ", inputs.attrs['contrast'])
```

Comment:
or in a debug mode?

Reply:
Since I never really read those print I simply removed them :)!